azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachineExtension.java

```diff
@@ -0,0 +1,546 @@
+         */
+        interface WithAttach<ParentT> extends
+                Attachable.InUpdate<ParentT>,
+                WithAutoUpgradeMinorVersion<ParentT>,
+                WithSettings<ParentT>,
+                WithTags<ParentT> {
+        }
+
+        /**
+         * The stage of the virtual machine extension definition allowing to enable or disable auto upgrade of the
+         * extension when when a new minor version of virtual machine extension image gets published.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithAutoUpgradeMinorVersion<ParentT> {
+            /**
+             * enables auto upgrade of the extension.
+             *
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withAutoUpgradeMinorVersionEnabled();
+
+            /**
+             * disables auto upgrade of the extension.
+             *
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withAutoUpgradeMinorVersionDisabled();
+        }
+
+        /**
+         * The stage of the virtual machine extension definition allowing to specify the public and private settings.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithSettings<ParentT> {
+            /**
+             * Specifies a public settings entry.
+             *
+             * @param key the key of a public settings entry
+             * @param value the value of the public settings entry
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withPublicSetting(String key, Object value);
+
+            /**
+             * Specifies a private settings entry.
+             *
+             * @param key the key of a private settings entry
+             * @param value the value of the private settings entry
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withProtectedSetting(String key, Object value);
+
+            /**
+             * Specifies public settings.
+             *
+             * @param settings the public settings
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withPublicSettings(HashMap<String, Object> settings);
+
+            /**
+             * Specifies private settings.
+             *
+             * @param settings the private settings
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withProtectedSettings(HashMap<String, Object> settings);
+        }
+
+        /**
+         * The stage of the virtual machine extension definition allowing to specify the tags.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithTags<ParentT> {
```

Comment:
The only constrain that base types  `ExternalChildResource` &  `ExtenralChildResourceImpl` enforces on the derived type is, the derived class requires an `id()`. Tags are not moved to the base because there are external child resources without tags e.g. Subnets.

So are you suggesting to introduce a new types something like  `TaggableExternalChildResource` & `TaggableExternalChildResourceImpl`?